### PR TITLE
add exception handling for lab not found in masterList

### PR DESCRIPTION
**Pull request by geetengtan**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/FilterCommand.java [hunk 1]
```diff
@@ -27,8 +31,12 @@ public FilterCommand(StudentHasLabPredicate predicate) {
     }
 
     @Override
-    public CommandResult execute(Model model) {
+    public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+        Lab toFilter = predicate.getLab();
+        if (!model.hasLab(toFilter)) {
+            throw new CommandException(String.format(MESSAGE_LAB_NOT_FOUND, toFilter.labNumber));
+        }
```
Comment (bryans17) [inline src/main/java/seedu/address/logic/commands/FilterCommand.java:39]: Just to confirm that you're trying to check if a lab is in the system based on lab number only. The `hasLab` function uses `isSameLab` which only checks the lab number. Just checking if this was intentional or you had something else in mind 🙃

**Review by bryans17 — COMMENTED** [review]
LGTM 👍🏻 . But just wanted to check if the addition was as intended.

File: src/main/java/seedu/address/logic/commands/FilterCommand.java (hunk 1 above)
Comment (geetengtan) [reply src/main/java/seedu/address/logic/commands/FilterCommand.java:39]: Yeap, i think only need to check lab number here.

**Review by bryans17 — APPROVED** [review]
LGTM

**Pull request merged**